azkustodata/keywords/keywords.go

```diff
@@ -0,0 +1,133 @@
+	Streaming                                      = "Streaming"
+	Uncompressed                                   = "Uncompressed"
+	EnforceMfa                                     = "EnforceMfa"
+	Accept                                         = "Accept"
+	QueryConsistency                               = "Query Consistency"
+	DataSourceUri                                  = "Data Source Uri"
+	AzureRegion                                    = "Azure Region"
+	Namespace                                      = "Namespace"
+	ApplicationCertificateIssuerDistinguishedName  = "Application Certificate Issuer Distinguished Name"
+	ApplicationCertificateSubjectDistinguishedName = "Application Certificate Subject Distinguished Name"
+)
+
+var (
+	Instance = createInstance()
+	//go:embed kcsb.json
+	jsonFile []byte
+)
+
+var (
+	supportedKeywords = map[string]bool{
+		DataSource:                true,
+		InitialCatalog:            true,
+		FederatedSecurity:         true,
+		ApplicationClientId:       true,
+		ApplicationKey:            true,
+		UserId:                    true,
+		AuthorityId:               true,
+		ApplicationToken:          true,
+		UserToken:                 true,
+		ApplicationCertificateX5C: true,
+		ApplicationNameForTracing: true,
+		UserNameForTracing:        true,
+		Password:                  true,
+
+		ApplicationCertificateBlob:       false,
+		ApplicationCertificateThumbprint: false,
+		DstsFederatedSecurity:            false,
+		Streaming:                        false,
+		Uncompressed:                     false,
+		EnforceMfa:                       false,
+		Accept:                           false,
+		QueryConsistency:                 false,
+		DataSourceUri:                    false,
+		AzureRegion:                      false,
+		Namespace:                        false,
+		ApplicationCertificateIssuerDistinguishedName:  false,
+		ApplicationCertificateSubjectDistinguishedName: false,
+	}
+	lookup = map[string]Keyword{}
+)
+
+// Keyword represents an individual keyword in the JSON.
+type Keyword struct {
+	Name        string   `json:"name"`
+	Aliases     []string `json:"aliases"`
+	Type        string   `json:"type"`
+	Secret      bool     `json:"secret"`
+	IsSupported bool
+}
+
+// Config represents the root structure of the JSON.
+type Config struct {
+	Version  string    `json:"version"`
+	Keywords []Keyword `json:"keywords"`
+}
+
+// normalizeKeyword normalizes a keyword by making it lowercase and removing spaces.
+func normalizeKeyword(keyword string) string {
+	return strings.ReplaceAll(strings.ToLower(keyword), " ", "")
+}
+
+func createInstance() *Config {
+	config := Config{}
+	err := json.Unmarshal(jsonFile, &config)
+	if err != nil {
+		panic(err.Error())
+	}
+
+	for _, word := range config.Keywords {
+		supported, ok := supportedKeywords[word.Name]
+		if !ok {
+			panic("Keyword " + word.Name + " is unexpected. This is a bug - please report it.")
```

Comment:
This means that the json and code don't align.
Maybe we should leave a dev oriented message here and possibly a comment as to what to do